### PR TITLE
Asset refactor

### PR DESCRIPTION
**Pull request by ericnewcomer**
* Remove groups, contacts, and flows from redux
* Remove SearchResult in favor of consistent use of Asset
* Uniform item storage in AssetService

Next Step: 
This temporarily disables FlowList. The next step will be to reimplement that as a simple control outside the editor backed by the AssetService (can be it's own).

### CODE REVIEW

File: src/services/AssetService.ts [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+    type: string;
+
+    isNew?: boolean;
+    content?: any;
+}
 
 export class Assets {
     private endpoint: string;
     private localStorage: boolean;
     protected nameProperty: string;
```
Comment (ycleptkellan) [inline src/services/AssetService.ts]: should we be typing `nameProperty`, `idProperty` and `assetType`? we've used Enums in the past, curious whether you think thats good practice or not

File: src/services/AssetService.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
 // tslint:disable:max-classes-per-file
-import { SearchResult } from '../store';
 import { Endpoints, FlowEditorConfig, ContactProperties, AttributeType } from '../flowTypes';
 import axios, { AxiosResponse } from 'axios';
 import { FlowComponents } from '../store/helpers';
+import { dump } from '../utils';
+
+export interface Asset {
```
Comment (ycleptkellan) [inline src/services/AssetService.ts]: just a head's up, `Simulator` has an `Asset` interface as well

File: src/services/AssetService.ts (hunk 1 above)
Comment (ericnewcomer) [reply src/services/AssetService.ts]: agreed, enum makes more sense here

File: src/services/AssetService.ts (hunk 2 above)
Comment (ericnewcomer) [reply src/services/AssetService.ts]: not for long it doesn't! simulator is going to need a bit of reworking for assetage as well, imagine it won't need it's own thing.

**Comment by ericnewcomer** [comment]
Coverage dropped due to some stuff I needed to comment out, I'll make good in the next PR, I promise!

File: src/services/AssetService.ts [hunk 3]
```diff
@@ -21,34 +28,49 @@ export class Assets {
...
+            const url = `${this.endpoint}?uuid=${id}`;
             axios
                 .get(url)
                 .then((response: AxiosResponse) => {
                     const ob = response.data.results[0];
+                    const asset = {
+                        id: ob.uuid,
+                        name: ob.name,
+                        type: this.assetType,
+                        content: ob.definition
```
Comment (ycleptkellan) [inline src/services/AssetService.ts]: just curious, why might a flow asset also need its definition?

Comment (ericnewcomer) [reply src/services/AssetService.ts]: this is the case of fetching by uuid and getting the full definition

Comment (ycleptkellan) [reply src/services/AssetService.ts]: ah, doh, thanks

**Pull request merged**